lib/sycamore/sycamore/connectors/weaviate/weaviate_writer.py

```diff
@@ -189,6 +192,10 @@ def from_doc(cls, document: Document, target_params: BaseDBWriter.TargetParams)
             "shingles": document.shingles,
         }
         droperties = drop_types(properties, drop_empty_lists=True)
+        if target_params.flatten_properties:
+            # Property names must be [a-zA-Z][_0-9a-zA-Z]{0,230}, so use __ as a separator rather than .
+            droperties = dict(flatten_data(droperties, allowed_list_types=[int, str, bool, float], separator="__"))
```

Comment:
Any reason we don't allow for `bytes` or `complex` data types here?

Reply:
not supported by weaviate: https://weaviate.io/developers/weaviate/config-refs/datatypes